master/static/migrations/20220311095512_add_workspace_tables.up.sql

```diff
@@ -0,0 +1,25 @@
+CREATE TABLE workspaces (
+  id SERIAL PRIMARY KEY,
+  name TEXT UNIQUE NOT NULL,
+  archived BOOLEAN,
+  user_id INT REFERENCES users(id)
+);
+CREATE TABLE projects (
+  id SERIAL PRIMARY KEY,
+  name TEXT UNIQUE NOT NULL,
+  description TEXT,
+  archived BOOLEAN,
+  workspace_id INT REFERENCES workspaces(id),
+  user_id INT REFERENCES users(id)
+);
+CREATE TABLE notes (
+  id SERIAL PRIMARY KEY,
+  name TEXT NOT NULL,
+  contents TEXT,
+  project_id INT REFERENCES projects(id)
+);
+CREATE TABLE project_models (
+  name TEXT,
+  checkpoint_id INT
+);
```

Comment:
**question:** this table seems like it's missing a couple of things. I can't remember the full context but some that I can think of are:
```
project_id INT REFERENCES projects(id)
creation_time timestamp with time zone NOT NULL
```

Reply:
added these in, plus an index for the project_id